### PR TITLE
Plasma dispersion function added to analytic_functions

### PR DESCRIPTION
**Pull request by namurphy**
My major changes are:
- Creating `plasma_dispersion_func` and `plasma_dispersion_func_deriv` in analytic_functions, with detailed docstrings and tests
- Created `tests` subdirectories in folders with empty __init__.py files for organization
- Added a warning to top-level __init__.py that says that PlasmaPy does not guarantee support for Python versions older than 3.6
Any suggestions are appreciated!

### CODE REVIEW

**Comment by namurphy** [comment]
Also I'm pretty sure I'm the only person in the last 17 years who has checked out this copy of "The Plasma Dispersion Function: The Hilbert Transform of the Gaussian" from my local astronomy library...

File: plasmapy/analytic_functions/tests/test_dispersion.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+def test_plasma_dispersion_func():
+    """Test the implementation of plasma_dispersion_func against exact
+    results, quantities calculated by Fried & Conte (1961), and
+    analytic results.
+    """
+
+    atol = 1e-10*(1+1j)
+
+    assert np.isclose(plasma_dispersion_func(0), 1j*np.sqrt(np.pi), \
```
Comment (SolarDrew) [inline plasmapy/analytic_functions/tests/test_dispersion.py]: You can drop the `\` inside parentheses.

**Comment by SolarDrew** [comment]
I'm getting test failures for plasma_dispersion_func(1) and plasma_dispersion_func_deriv(1).

**Comment by namurphy** [comment]
Oh, grumblemuffins.  I'll get to this when I have a chance.  Thanks for noticing that!  I need to do the formatting update you mentioned too.

**Comment by SolarDrew** [comment]
No problem. Also I think eb36a7201933a7da1121f1b665051cda237c9c53, which removes these files, is being applied over this PR and negating it.

**Comment by Cadair** [comment]
You need some CI ;)

**Comment by Cadair** [comment]
I mean, you would have to give me write access to the repo... or one of you can turn travis on and i can try and do the rest.

**Pull request merged**